openmls/src/key_packages/mod.rs

```diff
@@ -323,29 +325,6 @@ impl KeyPackage {
         &self,
         backend: &impl OpenMlsCryptoProvider,
     ) -> Result<(), KeyPackageVerifyError> {
-        // Extension included in the extensions or leaf_node.extensions fields
```

Comment:
These (and other) validations could all be deleted when there is already a (always validated) `LeafNode<KeyPackage>`.